language_service/src/test_utils.rs

```diff
@@ -100,44 +70,47 @@ pub(crate) fn compile_with_fake_stdlib(source_name: &str, source_contents: &str)
     }
 }
 
-pub(crate) fn compile_notebook_with_fake_stdlib_and_markers(
-    cells_with_markers: &[(&str, &str)],
+pub(crate) fn compile_project_with_fake_stdlib_and_markers(
```

Comment:
Isn't the usual pattern to get marker offsets with `get_source_and_marker_offsets` first, then call into `compile_with_fake_stdlib`? Why do we need this function?

Reply:
Because there are multiple files, and it's nontrivial to iterate over them, extract markers, validate that there is only one cursor offset, etc. I thought it was worth encapsulating all of that in a utility.

You're right that now there's an asymmetry between this and _original_ way of compiling a single file ... I was thinking I would extract the single-file compilation into a helper method too, but that would involve touching all the test files. I'm happy to do it in this PR if you don't mind reviewing ;)

Reply:
Yah I think that might be the right way to go. It will make for a bigger PR, but I'd rather have consistency in how the tests work.